src/expire.c

```diff
@@ -521,6 +569,45 @@ void expireGenericCommand(client *c, long long basetime, int unit) {
         return;
     }
 
+    /* NX option is set, check current expiry */
+    if (flag & FLAG_NX) {
+        current_expire = getExpire(c->db, key);
+        if (current_expire != -1) {
+            addReply(c,shared.czero);
+            return;
+        }
+    }
+
+    /* XX option is set, check current expiry */
+    if (flag & FLAG_XX) {
+        current_expire = getExpire(c->db, key);
+        if (current_expire == -1) {
+            /* reply 0 when the key has no expiry */
+            addReply(c,shared.czero);
+            return;
+        }
+    }
+
+    /* GT option is set, check current expiry */
+    if (flag & FLAG_GT) {
+        current_expire = getExpire(c->db, key);
+        if (when <= current_expire || current_expire == -1) {
```

Comment:
trying to follow the logic (maybe a comment is needed)...
if the key doesn't have expire time, it's as is it has an infinite expire time (expires in the far far future), we consider that as if our new expire time is **not** grater than that infinite number.
right?

p.s. we do have a test that covers that, right?

Reply:
Yes and there is a test for this case: https://github.com/redis/redis/pull/2795/files#diff-791d7b1ec2ea62ae57b4b424a8c50b02736afa4a75c0c2ae38939031ee92cd2eR675